File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/SubstanceNucleicAcid.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.Resource;

public class SubstanceNucleicAcid : DomainResource, IResource
{
    public SubstanceNucleicAcid_Subunit[]? Subunit { get; set; }
    public Base.CodeableConcept? SequenceType { get; set; }
    public int? NumberOfSubunits { get; set; }
    public string? AreaOfHybridisation { get; set; }
    public Base.CodeableConcept? OligoNucleotideType { get; set; }

    public class SubstanceNucleicAcid_Subunit_Sugar : BackboneElement
    {
        public string? Name { get; set; }
        public Base.Identifier? Identifier { get; set; }
        public string? ResidueSite { get; set; }
    }

    public class SubstanceNucleicAcid_Subunit_Linkage : BackboneElement
    {
        public string? Name { get; set; }
        public Base.Identifier? Identifier { get; set; }
        public string? ResidueSite { get; set; }
        public string? Connectivity { get; set; }
    }

    public class SubstanceNucleicAcid_Subunit : BackboneElement
    {
        public SubstanceNucleicAcid_Subunit_Sugar[]? Sugar { get; set; }
        public int? Length { get; set; }
        public SubstanceNucleicAcid_Subunit_Linkage[]? Linkage { get; set; }
        public int? Subunit { get; set; }
        public string? Sequence { get; set; }
        public Base.CodeableConcept? FivePrime { get; set; }
        public Base.CodeableConcept? ThreePrime { get; set; }
        public Base.Attachment? SequenceAttachment { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/SubstanceNucleicAcid.py
from typing import Optional, List
from pydantic import *
from ..base.Attachment import Attachment
from ..base.CodeableConcept import CodeableConcept
from ..base.DomainResource import DomainResource
from ..base.Identifier import Identifier
from ..base.BackboneElement import BackboneElement

class SubstanceNucleicAcid_Subunit_Sugar(BackboneElement):
    name: Optional[str] = None
    identifier: Optional[Identifier] = None
    residue_site: Optional[str] = None

class SubstanceNucleicAcid_Subunit_Linkage(BackboneElement):
    name: Optional[str] = None
    identifier: Optional[Identifier] = None
    residue_site: Optional[str] = None
    connectivity: Optional[str] = None

class SubstanceNucleicAcid_Subunit(BackboneElement):
    sugar: Optional[List[SubstanceNucleicAcid_Subunit_Sugar]] = None
    length: Optional[int] = None
    linkage: Optional[List[SubstanceNucleicAcid_Subunit_Linkage]] = None
    subunit: Optional[int] = None
    sequence: Optional[str] = None
    five_prime: Optional[CodeableConcept] = None
    three_prime: Optional[CodeableConcept] = None
    sequence_attachment: Optional[Attachment] = None

class SubstanceNucleicAcid(DomainResource):
    subunit: Optional[List[SubstanceNucleicAcid_Subunit]] = None
    sequence_type: Optional[CodeableConcept] = None
    number_of_subunits: Optional[int] = None
    area_of_hybridisation: Optional[str] = None
    oligo_nucleotide_type: Optional[CodeableConcept] = None